### ISSUE TITLE
Windows: Neovide 0.12 regressions

### ISSUE THREAD

**Issue by VladimirMarkelov** [neovide/neovide#2201]
**Describe the bug**

Neovim `0.9.4` (Neovide 0.12 requires 0.9.2 or newer, so it must be fine).
Neovim is in the directory `d:\Utils\neovim\bin\`, I put the new Neovide version **into the same** directory.

Comparing to the previous version of Neovide, the new version has a few regressions:

- If I am not mistaken, there was such an issue with older versions and it was fixed: Neovide 0.12 parses the command line options and unescapes them. Example: since Neovide 0.11.2 I used a command `"d:\Utils\neovim\bin\neovide.exe" --neovim-bin="d:\utils\neovim\bin\nvim.exe" --geometry=140x45` - it worked. First, I had to change, as Neovide suggested, the command line to `"d:\Utils\neovim\bin\neovide.exe" --neovim-bin="d:\utils\neovim\bin\nvim.exe" -- --geometry=140x45` (added extra ` -- ` before `--geometry`). At the same time, Neovide 0.12 started requiring different kind of slashes: `d:\Utils\neovim\bin\neovide.exe" --neovim-bin="d:/utils/neovim/bin/nvim.exe" -- --geometry=140x45`. Only after all those changes I managed to start Neovide 0.12.
- But it was not the only issue. Now I have no clue how to pass `geometry`. If I do not add an extra `--`, Neovide complains that the extra options must be passed after `--`. When I add ` -- ` before `--geometry`, Neovim starts with a default tiny window. How to set window size now?
- The last issue: since 0.12 Neovide started complaining about unknown options in `ginit.nvim`. It contains commands for `neovim-qt`, like `call GuiWindowMaximized(1)` and `GuiFont!`. Neovide 0.12 complains at both command, so I had to comment them out, because the way I use for `init.vim` does not work here. I tried to add this:

```
if exists("g:nvy")
else
	call GuiWindowMaximized(1)
	GuiFont! JuliaMono:h11
endif
```

But Neovide keeps complaining about non-existing functions. How to mark commands in `ginit.vim` so Neovide skips them?

**To Reproduce**
1. `cd d:\utils\neovim\bin`
2. `neovide` - works fine, but the windows is tiny 
3. `neovide --neovim-bin="d:\utils\neovim\bin\nvim.exe"` - crashes with:

![image](https://github.com/neovide/neovide/assets/1577186/972ef505-1066-4dc0-8479-6fbc70022f67)

Sorry for pasting the picture, but the suggested `y - copy` in the Windows does not copy anything to the clipboard. Maybe it is another issue under Windows.

5. After pressing `y`, the windows closes and a log file created:

```
2023-12-24 19:13:29 - Neovide panicked with the message 'called `Option::unwrap()` on a `None` value'. (File: src\clipboard.rs; Line: 36, Column: 21)
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: CallWindowProcW
  14: CallWindowProcW
  15: glPushClientAttrib
  16: CallWindowProcW
  17: DispatchMessageW
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: BaseThreadInitThunk
  25: RtlUserThreadStart
```

I am not sure if it is related to the error in the picture, but it is all I have.

**Expected behavior**

1.  setting the windows size with `--geometry`(or any other way) should work
2. On Windows it should not unescape the path passed in `--neovim-bin`
3. `if exists("g:nvy")` should work in `ginit.vim`. Or please provide another way to skip a block of functions for another Neovim GUI

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10 64 bit
 - Neovide Version 0.12
 - Neovim Version 0.9.4

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**

`"d:\Utils\neovim\bin\neovide_new.exe" --neovim-bin="D:\Utils\neovim\bin\nvim.exe" --log`

[neovide_new_r2023-12-24_11-32-24.log](https://github.com/neovide/neovide/files/13762675/neovide_new_r2023-12-24_11-32-24.log)

**Comment by VladimirMarkelov** [comment]
I found one more log that supports my version about redundant unescaping:

```
TRACE [neovide] Neovide version: 0.12.0
DEBUG [neovide::settings::window_size] Loaded window settings: Windowed { position: PhysicalPosition { x: 104, y: 104 }, pixel_size: Some(PhysicalSize { width: 500, height: 500 }), grid_size: Some(Dimensions { width: 55, height: 27 }) }
DEBUG [neovide::bridge::command] Running which command: "which" "D:Utilsneovimbinnvim.exe"
ERROR [neovide::error_handling] Neovide just crashed :(
This is the error that caused the crash. In case you don't know what to do with this, please feel free to report this on https://github.com/neovide/neovide/issues!

ERROR: NEOVIM_BIN='D:\Utils\neovim\bin\nvim.exe' was not found.
```

**Comment by fredizzimo** [comment]
I will comment more about the first two issues after some more investigation.

But regarding the third one, that's actually not a regression, but a bugfix, before it just silently ignored all errors in `ginit.vim` . I recommend that you follow the documentation of Neovim-qt https://github.com/equalsraf/neovim-qt?tab=readme-ov-file#configuration, and check if the functions exist.

**Comment by fredizzimo** [comment]
@VladimirMarkelov, do you have a logfile where you try to pass `--neovim-bin` the one you attached does not seem to do that.

Also which shell are you using `powershell`, `cmd` or something else?

**Comment by VladimirMarkelov** [comment]
@fredizzimo As I mentioned in my second comment, I found a very short log file with the error:
[neovide_new_rCURRENT.log](https://github.com/neovide/neovide/files/13767008/neovide_new_rCURRENT.log)

It shows that all slashes are gone from the file name before checking the binary existence.


Thanks a lot for the link to `neovim-qt`, I did not think I could do it this way.

**Comment by VladimirMarkelov** [comment]
> Also which shell are you using powershell, cmd or something else?

I run Neovide from a file manager - FAR that runs within ConEmu. I believe, internally ConEmu uses `cmd.exe` by default.

**Comment by fredizzimo** [comment]
This escaping problem will be fixed by #2210 

And the `--geometry` argument has been replaced by `--grid` and `--size`. I will document that as a breaking change in the release notes.

https://neovide.dev/command-line-reference.html#window-size
https://neovide.dev/command-line-reference.html#grid-size

**Comment by VladimirMarkelov** [comment]
That's great! With `--grid` and slashes instead of backslashes, I was able to run the new version of Neovide.

One question: I see Neovide's config contains `maximized` and `neovim-bin` options. Though, it does not provide a way to set default `grid` and `size` options. Do you plan to add `grid` and `size` options to `config.toml` as well?

**Comment by fredizzimo** [comment]
> One question: I see Neovide's config contains `maximized` and `neovim-bin` options. Though, it does not provide a way to set default `grid` and `size` options. Do you plan to add `grid` and `size` options to `config.toml` as well?

That would probably be good to add yes, but there are a couple of ways of doing that currently.

You can use `colums` and `lines` in your `init.lua/vim` instead of `--grid`, so that's one way to make them permanent. https://neovim.io/doc/user/options.html#'columns'

For the size, you can save the size once and then set https://neovide.dev/configuration.html#remember-previous-window-size to false, and not save it again.
